### DISCUSSION TITLE
Google OAuth API verification and consent screen

### DISCUSSION THREAD

**Discussion by taylormcknight** [nhost/nhost#983] [Q&A]
I'm trying to submit my web app for Google's OAuth API verification so that I can move from testing to production.

Google currently rejects my application because I (obviously) cannot verify domain ownership of nhost.run, the domain associated with the OAuth Callback URL.

Is there a way I can get the Google OAuth API verified while using nhost? Should I use my own domain and then redirect to the nhost OAuth Callback URL?

<img width="523" alt="Screen Shot 2022-10-02 at 1 14 12 PM" src="https://user-images.githubusercontent.com/4059515/193467100-0b0f4dd1-6284-40d7-9cde-23d029302986.png">

**Comment by dionjwa** [comment]
I have also hit this exact problem. 

If I try to redirect e.g. `google-oauth.<mydomain>` to the nhost google oauth redirect URL, it fails with 

<img width="671" alt="image" src="https://user-images.githubusercontent.com/260221/202943507-54bf11bb-f580-49dc-838d-c82deca08247.png">

**Comment by osseonews** [comment]
No answer here? There is no way to set up Google Auth with Nhost in production, because Nhost is redirecting the request thru nhost.run which is not a domain that can be verified, plus the user is expecting to go to the verified domain, not nhost.run. Why would a user go to a website and permit themselves to then login into nhost.run? This is obviously a massive security issue, even if you could possibly verify nhost.run, which you can't. How did anyone set up Google Social Auth with nhost?
